server/src/test/java/org/opensearch/action/search/SearchRequestOperationsListenerAssertingListener.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class SearchRequestOperationsListenerAssertingListener extends SearchRequestOperationsListener {
+    private volatile SearchPhase phase;
+
+    @Override
+    protected void onPhaseStart(SearchPhaseContext context) {
+        assertThat(phase, is(nullValue()));
+        phase = context.getCurrentPhase();
+    }
+
+    @Override
+    protected void onPhaseEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        assertThat(phase, is(context.getCurrentPhase()));
+        phase = null;
+    }
+
+    @Override
+    protected void onPhaseFailure(SearchPhaseContext context, Throwable cause) {
+        assertThat(phase, is(context.getCurrentPhase()));
+        phase = null;
+    }
```

Comment:
Could you please add the assertion method here, something like that:

```
public void assertFinished() {
    assertThat(phase, is(nullValue()));
}
```

We need to call this method in `@After` hooks to ensure that each phase is properly passed through the lifecycle / listener callbacks, thank you.

Reply:
Added